### PR TITLE
Fix test runner, smarter server

### LINKED ISSUE #167: Puppeteer tests seem to no longer work when run locally.
When trying to run tests locally via `npm test`, I am currently getting the following:

```
ProtocolError: Protocol error (Runtime.callFunctionOn): Execution context was destroyed.
    at <instance_members_initializer> (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:92:14)
    at new Callback (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:96:16)
    at CallbackRegistry.create (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/common/CallbackRegistry.js:19:26)
    at Connection._rawSend (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/cdp/Connection.js:77:26)
    at CdpCDPSession.send (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/cdp/CDPSession.js:63:33)
    at #evaluate (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/cdp/ExecutionContext.js:199:50)
    at ExecutionContext.evaluate (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/cdp/ExecutionContext.js:114:36)
    at IsolatedWorld.evaluate (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/cdp/IsolatedWorld.js:121:30)
    at CdpFrame.evaluate (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/api/Frame.js:350:43)
    at CdpFrame.<anonymous> (file:///Users/aseier@netflix.com/WebstormProjects/x-element/node_modules/puppeteer-core/lib/esm/puppeteer/util/decorators.js:94:27)
TAP version 14
1..0 # no tests found
✨  Done in 0.77s.
```

There are two problems here:

1. This should _not_ result in successful (`0`) command exit which hides the problem from tooling.
2. The tests seem to not actually run at this point, so we need to fix that.

Mostly (2) is the thing that needs to be fixed.

### PR DESCRIPTION
**Pull request by klebba**
Fixes #167

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -93,7 +93,10 @@ class Server {
...
 }
 
-http.createServer(Server.requestListener).listen(port, () => {
-  // eslint-disable-next-line no-console
-  console.log(`Development server running: http://localhost:${port}`);
+const server = http.createServer(Server.requestListener);
+
+server.listen(process.env.PORT || 8080, () => {
+  const { address, port } = server.address();
+  Server.origin = `http://[${address}]:${port}`;
```
Comment (klebba) [inline server.js:100]: We use introspection to generate the `origin` component necessary to qualify the path and leverage `URL` above for smarter path parsing.

File: server.js [hunk 2]
```diff
@@ -2,28 +2,28 @@ import http from 'http';
...
 
 class Server {
+
+  static origin;
+
   static requestListener(request, response) {
     if (request.method === 'GET') {
-      const fileUrl = request.url;
-      const filePath = path.resolve(`${root}/${fileUrl}`);
+      const url = new URL(request.url, Server.origin);
```
Comment (klebba) [inline server.js:25]: By adopting `URL` here we can easily extract `pathname` and fix the confusion around paths which incorporate query params. Easy peasy.

File: test.js [hunk 3]
```diff
@@ -3,7 +3,17 @@ import puppeteer from 'puppeteer';
 (async () => {
   try {
     // Open our browser.
-    const browser = await puppeteer.launch({ timeout: 10000 });
+    const browser = await puppeteer.launch({
```
Comment (klebba) [inline test.js:6]: These are just a few flags I borrowed from another project (Profiler)

File: test.js [hunk 4]
```diff
@@ -3,7 +3,17 @@ import puppeteer from 'puppeteer';
...
+      timeout: 10000,
+      // opt-in to the new Chrome headless implementation
+      // ref: https://developer.chrome.com/articles/new-headless/
+      headless: 'new',
```
Comment (theengineear) [inline test.js:10]: Oh, we using the new, new. Cool.

File: test.js [hunk 5]
```diff
@@ -3,7 +3,17 @@ import puppeteer from 'puppeteer';
...
+      args: [
+        // Disables interactive prompt: Do you want to the application Chromium.app to accept incoming network connections?
+        // ref: https://github.com/puppeteer/puppeteer/issues/4752#issuecomment-586599843
+        '--disable-features=DialMediaRouteProvider',
```
Comment (theengineear) [inline test.js:14]: 👌

**Review by theengineear — APPROVED** [review]
Thanks for the quick turnaround @klebba!

**Pull request merged**